### PR TITLE
[SYNCOPE-1557] complete policies manipulating

### PR DESCRIPTION
**Pull request by DimaAy**
remove

ITCase attempt

Cleanup + sample working test for ClientApp

SyncopeServiceRegistry tests

[SYNCOPE-1557] authModules  manipulating

[SYNCOPE-1557] complete policies manipulating and adding more  tests

### CODE REVIEW

File: common/am/lib/src/main/java/org/apache/syncope/common/lib/wa/WAClientApp.java [hunk 1]
```diff
@@ -36,6 +38,8 @@
 
     private AttrReleasePolicyConf attrReleasePolicyConf;
 
+    private final Map<String, Object> releaseAttributes = new HashMap<>();
```
Comment (ilgrosso) [inline common/am/lib/src/main/java/org/apache/syncope/common/lib/wa/WAClientApp.java:41]: please rename as `releaseAttrs` to follow the conventions

File: common/am/lib/src/main/java/org/apache/syncope/common/lib/wa/WAClientApp.java [hunk 2]
```diff
@@ -67,4 +71,8 @@ public AttrReleasePolicyConf getAttrReleasePolicyConf() {
     public void setAttrReleasePolicyConf(final AttrReleasePolicyConf attrReleasePolicyConf) {
         this.attrReleasePolicyConf = attrReleasePolicyConf;
     }
+
+    public Map<String, Object> getReleaseAttributes() {
```
Comment (ilgrosso) [inline common/am/lib/src/main/java/org/apache/syncope/common/lib/wa/WAClientApp.java:75]: Same as above

File: core/provisioning-java/src/main/java/org/apache/syncope/core/provisioning/java/data/wa/WAClientAppBinderImpl.java [hunk 3]
```diff
@@ -35,15 +39,21 @@
...
     @Override
     public WAClientApp getWAClientApp(final ClientApp clientApp) {
         WAClientApp waClientApp = new WAClientApp();
         waClientApp.setClientAppTO(clientAppDataBinder.getClientAppTO(clientApp));
 
         try {
+            AuthPolicyConf authPolicyConf = null;
             if (clientApp.getAuthPolicy() != null) {
+                authPolicyConf = clientApp.getAuthPolicy().getConf();
                 waClientApp.setAuthPolicyConf((clientApp.getAuthPolicy()).getConf());
```
Comment (ilgrosso) [inline core/provisioning-java/src/main/java/org/apache/syncope/core/provisioning/java/data/wa/WAClientAppBinderImpl.java:54]: this line must be removed

File: core/provisioning-java/src/main/java/org/apache/syncope/core/provisioning/java/data/wa/WAClientAppBinderImpl.java [hunk 4]
```diff
@@ -35,15 +39,21 @@
...
             } else if (clientApp.getRealm().getAuthPolicy() != null) {
+                authPolicyConf = clientApp.getAuthPolicy().getConf();
                 waClientApp.setAuthPolicyConf((clientApp.getRealm().getAuthPolicy()).getConf());
```
Comment (ilgrosso) [inline core/provisioning-java/src/main/java/org/apache/syncope/core/provisioning/java/data/wa/WAClientAppBinderImpl.java:57]: this line must be moved right after the subsequent `}`

File: core/provisioning-java/src/main/java/org/apache/syncope/core/provisioning/java/data/wa/WAClientAppBinderImpl.java [hunk 5]
```diff
@@ -58,6 +68,26 @@ public WAClientApp getWAClientApp(final ClientApp clientApp) {
             } else if (clientApp.getRealm().getAttrReleasePolicy() != null) {
                 waClientApp.setAttrReleasePolicyConf((clientApp.getRealm().getAttrReleasePolicy()).getConf());
             }
+
+            if (authPolicyConf != null && authPolicyConf instanceof DefaultAuthPolicyConf
```
Comment (ilgrosso) [inline core/provisioning-java/src/main/java/org/apache/syncope/core/provisioning/java/data/wa/WAClientAppBinderImpl.java:72]: Please remove `authPolicyConf != null &&` - `instanceof` will do the job

File: docker/wa/src/main/resources/application.properties [hunk 6]
```diff
@@ -26,8 +26,17 @@ spring.http.encoding.force=true
 
 server.servlet.contextPath=/syncope-wa
 
-spring.resources.static-locations=classpath:/syncope/static,classpath:/static
-spring.thymeleaf.templateResolverOrder=1 
+spring.resources.static-locations=classpath:/thymeleaf/static,classpath:/static
```
Comment (ilgrosso) [inline docker/wa/src/main/resources/application.properties:29]: Can you please check that
* `docker/wa/src/main/resources/application.properties`
* `docker/wa/src/main/resources/wa.properties`

are aligned with

* `wa/starter/src/main/resources/application.properties`
* `wa/starter/src/main/resources/wa.properties`

?

File: core/provisioning-java/src/main/java/org/apache/syncope/core/provisioning/java/data/wa/WAClientAppBinderImpl.java (hunk 4 above)
Comment (DimaAy) [reply core/provisioning-java/src/main/java/org/apache/syncope/core/provisioning/java/data/wa/WAClientAppBinderImpl.java:57]: ah, my bad this line has to be:
` authPolicyConf=(clientApp.getRealm().getAuthPolicy()).getConf(); `
as a sequence, authPoliceConf is needed in both condition

File: core/provisioning-java/src/main/java/org/apache/syncope/core/provisioning/java/data/wa/WAClientAppBinderImpl.java [hunk 7]
```diff
@@ -53,7 +53,7 @@ public WAClientApp getWAClientApp(final ClientApp clientApp) {
                 authPolicyConf = clientApp.getAuthPolicy().getConf();
                 waClientApp.setAuthPolicyConf((clientApp.getAuthPolicy()).getConf());
             } else if (clientApp.getRealm().getAuthPolicy() != null) {
-                authPolicyConf = clientApp.getAuthPolicy().getConf();
+                authPolicyConf = (clientApp.getRealm().getAuthPolicy()).getConf();
```
Comment (ilgrosso) [inline core/provisioning-java/src/main/java/org/apache/syncope/core/provisioning/java/data/wa/WAClientAppBinderImpl.java:56]: please remove the unneeded additional parenthesis

**Comment by mmoayyed** [comment]
Sure. I'll do a sanity check.

**Pull request merged**